File: application/src/main/java/de/wuespace/telestion/project/daedalus2/redis/RedisLatestRequest.java
package de.wuespace.telestion.project.daedalus2.redis;

import com.fasterxml.jackson.annotation.JsonProperty;
import com.mongodb.lang.NonNull;
import de.wuespace.telestion.api.message.JsonMessage;

import java.util.List;

@SuppressWarnings("unused")
public record RedisLatestRequest(
		@NonNull
		@JsonProperty List<String> fields
) implements JsonMessage {
	public RedisLatestRequest() {
		this(List.of());
	}
}

File: client/src/widgets/hooks/redis/model/redis-latest-request.ts
import { JavaMessage } from '../../../../model/java-message';

/**
 * Counterpart to `de.wuespace.telestion.project.daedalus2.redis.RedisLatestRequest` Java class
 */
export interface RedisLatestRequest extends JavaMessage {
	fields: string[];
	className: 'de.wuespace.telestion.project.daedalus2.redis.RedisLatestRequest';
}
